### PR TITLE
Store service configs in Secrets

### PR DESCRIPTION
**Pull request by gibizer**
The generated service configs containing sensitive data like database password. So we switch the config generation code to store them in Secrets instead of ConfigMaps.

Related: https://issues.redhat.com/browse/OSPRH-98

### CODE REVIEW

**Comment by gibizer** [comment]
This can be deployed and I can show that nova-operator now store things in Secrets. 
```
$ oc get -n openstack cm | grep nova
$ oc get -n openstack secret | grep nova
nova-api-config-data                                   Opaque                                4      119s
nova-cell0-conductor-config-data                       Opaque                                3      2m19s
nova-cell0-conductor-scripts                           Opaque                                1      2m19s
nova-cell0-manage-config-data                          Opaque                                3      119s
nova-cell0-manage-scripts                              Opaque                                1      119s
nova-cell1-conductor-config-data                       Opaque                                3      119s
nova-cell1-conductor-scripts                           Opaque                                1      119s
nova-cell1-manage-config-data                          Opaque                                3      100s
nova-cell1-manage-scripts                              Opaque                                1      100s
nova-metadata-config-data                              Opaque                                4      119s
nova-nova-dockercfg-szh9f                              kubernetes.io/dockercfg               1      3m22s
nova-nova-token-6xrbw                                  kubernetes.io/service-account-token   4      3m22s
nova-scheduler-config-data                             Opaque                                3      119s
rabbitmq-transport-url-nova-api-transport              Opaque                                1      2m30s
rabbitmq-transport-url-nova-cell1-transport            Opaque                                1      2m30s
```

**Comment by gibizer** [comment]
/test nova-operator-build-deploy-tempest
Mutliple controller managers in crashloop.

**Comment by gibizer** [comment]
/test nova-operator-build-deploy-tempest
Multiple test timed out

**Comment by gibizer** [comment]
rebased to re-use the already got secret.

**Review by mrKisaoLamb — APPROVED** [review]
lgtm

**Comment by gibizer** [comment]
/hold
The functional test failure is relevant...

File: test/functional/nova_controller_test.go [hunk 1]
```diff
@@ -244,29 +244,30 @@ var _ = Describe("Nova controller", func() {
...
-			mappingJobConfig := th.GetConfigMap(
+			mappingJobConfig := th.GetSecret(
 				types.NamespacedName{
 					Namespace: cell0.CellName.Namespace,
 					Name:      fmt.Sprintf("%s-config-data", cell0.CellName.Name+"-manage"),
 				},
 			)
 			Expect(mappingJobConfig.Data).Should(HaveKey("01-nova.conf"))
-			Expect(mappingJobConfig.Data["01-nova.conf"]).To(
+			configData := string(mappingJobConfig.Data["01-nova.conf"])
```
Comment (SeanMooney) [inline test/functional/nova_controller_test.go:254]: i assume you are pulling this out to do the type converion once and also avoid remeating the subscript ["01-nova.conf"]

you basicaly have the same change throught the file.

**Comment by gibizer** [comment]
> /hold The functional test failure is relevant...

It is a very strange test error:
```
  Full Stack Trace
    github.com/openstack-k8s-operators/lib-common/modules/test/helpers.(*TestHelper).GetConfigMap(0xc000dae7d0, {{0xc001900030?, 0xc0015de210?}, {0xc0008b6180?, 0x4e6b55?}})
    	/go/pkg/mod/github.com/openstack-k8s-operators/lib-common/modules/test@v0.0.0-20230619102827-49e72f626a11/helpers/configmap.go:34 +0x1b8
    github.com/openstack-k8s-operators/nova-operator/test/functional_test.glob..func6.6.1.1({0x1fcf638, 0xc0015de210})
    	/go/src/github.com/openstack-k8s-operators/nova-operator/test/functional/nova_reconfiguration_test.go:463 +0x4a```

* there is no GetConfigMap call in nova_reconfiguration_test.go
* there is no L 463 in nova_reconfiguration_test.go the file is 447 line long

**Comment by gibizer** [comment]
> > /hold The functional test failure is relevant...
> 
> It is a very strange test error:
> 
> ```
>   Full Stack Trace
>     github.com/openstack-k8s-operators/lib-common/modules/test/helpers.(*TestHelper).GetConfigMap(0xc000dae7d0, {{0xc001900030?, 0xc0015de210?}, {0xc0008b6180?, 0x4e6b55?}})
>     	/go/pkg/mod/github.com/openstack-k8s-operators/lib-common/modules/test@v0.0.0-20230619102827-49e72f626a11/helpers/configmap.go:34 +0x1b8
>     github.com/openstack-k8s-operators/nova-operator/test/functional_test.glob..func6.6.1.1({0x1fcf638, 0xc0015de210})
>     	/go/src/github.com/openstack-k8s-operators/nova-operator/test/functional/nova_reconfiguration_test.go:463 +0x4a
> ```
> 
>     * there is no GetConfigMap call in nova_reconfiguration_test.go
> 
>     * there is no L 463 in nova_reconfiguration_test.go the file is 447 line long

OMG. prow does speculative test too?! I rebased the patch locally to the recent main and now I see line 463 and the failure

**Comment by gibizer** [comment]
> OMG. prow does speculative test too?! I rebased the patch locally to the recent main and now I see line 463 and the failure

I did the rebase and adapted the incoming new test case.

**Comment by gibizer** [comment]
/retest
Infra issue
```
+ sleep 30
++ get_build_status nova-operator
+++ oc get builds -l buildconfig=nova-operator -o json
+++ jq -r '.items[0].status.phase'
The connection to the server api.oko-01.oooci.ccitredhat.com:6443 was refused - did you specify the right host or port?
```

**Comment by gibizer** [comment]
/test nova-operator-build-deploy
There was no test cluster to claim

**Review by SeanMooney — APPROVED** [review]
its not quite specultive mergeing in the depend on case but it does rebase the patch on the target branch when its testing it yes.

that's kind of the min speculation it must do to have the test be valid.

**Comment by gibizer** [comment]
needed another rebase on the newest main

**Comment by gibizer** [comment]
/test nova-operator-build-deploy
Slow node?

**Pull request merged**